libsolidity/formal/SolverInterface.h

```diff
@@ -65,16 +65,18 @@ struct FunctionSort: public Sort
 {
 	FunctionSort(std::vector<SortPointer> _domain, SortPointer _codomain):
 		Sort(Kind::Function), domain(std::move(_domain)), codomain(std::move(_codomain)) {}
-	bool operator==(FunctionSort const& _other) const
+	bool operator==(Sort const& _other) const override
 	{
+		auto _otherFunction = dynamic_cast<FunctionSort const*>(&_other);
+		solAssert(_otherFunction, "");
```

Comment:
Shouldn't it just return false if the other is not a function?

Reply:
Or start the function with `Sort::operator==(_other)`.

Reply:
Yea, that makes sense